src/lsptoolshost/nestedCodeAction.ts

```diff
@@ -0,0 +1,122 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as RoslynProtocol from './roslynProtocol';
+import { CodeAction, CodeActionResolveRequest, LSPAny } from 'vscode-languageserver-protocol';
+import { RoslynLanguageServer } from './roslynLanguageServer';
+import { URIConverter, createConverter } from 'vscode-languageclient/lib/common/protocolConverter';
+import { UriConverter } from './uriConverter';
+
+export function registerNestedCodeActionCommands(
+    context: vscode.ExtensionContext,
+    languageServer: RoslynLanguageServer,
+    outputChannel: vscode.OutputChannel
+) {
+    context.subscriptions.push(
+        vscode.commands.registerCommand(
+            'roslyn.client.nestedCodeAction',
+            async (request): Promise<void> => registerNestedResolveCodeAction(languageServer, request, outputChannel)
+        )
+    );
+}
+
+async function registerNestedResolveCodeAction(
+    languageServer: RoslynLanguageServer,
+    codeActionData: any,
+    outputChannel: vscode.OutputChannel
+) {
+    if (codeActionData) {
+        const data = <LSPAny>codeActionData;
+        const action = data.NestedCodeAction;
+
+        if (action && action.nestedActions && action.nestedActions.length > 0) {
```

Comment:
This could be simplified.
```suggestion
        if (action?.nestedActions?.length > 0) {
```